app/src/main/java/com/jvoye/tasky/agenda_detail/presentation/AgendaDetailScreenViewModel.kt

```diff
@@ -41,6 +47,20 @@ class AgendaDetailScreenViewModel(
             taskyItem = agendaRepository.getTaskyItem(taskyItemId)
         ) }
     }
+
+    private fun getNavArgs() {
+        val savedState =  savedStateHandle.get<TaskyNavParcelable>("TaskyNavArgs")
+
+        _state.update { it.copy(
+            taskyItemId = savedState?.taskyItemId,
+            taskyItemType = savedState?.taskyItemType ?: TaskyType.TASK,
+            isEdit = savedState?.isEditMode ?: true
+
+        ) }
```

Comment:
Minor: All these seem to be values that never change across the lifetime of this ViewModel. In that case they wouldn't have to be part of the state (since the definition of state is literally "a value that can change over time"). 

It's not a big deal however, if you like having it part of it, that's fine 😄